src/awkward/highlevel.py

```diff
@@ -181,18 +183,6 @@ def __init__(
             layout = data._layout
             behavior = ak._util.behavior_of(data, behavior=behavior)
 
-        elif numpy.is_own_array(data) and data.dtype != np.dtype("O"):
```

Comment:
Now this logic goes through `to_layout` after the addition of a new `regulararray` argument.